docs/src/main/sphinx/connector/filesystem-cache.md

```diff
@@ -0,0 +1,115 @@
+cache. Performance gains are more significant if the performance of directly
+accessing the storage is low compared to accessing the local cache.
+
+For example, if you access storage in a different network, different data
+center, or even different cloud-provider region query performance is slow. Adding
+caching using fast, local storage has a significant impact and makes your
+queries much faster.
+
+On the other hand, if your storage is already running at very high performance
+for I/O and network access, and your local cache storage is at similar speeds,
+or even slower, performance benefits can be minimal.
+
+**Reduced query costs**
+
+A result of the reduced load on the storage, mentioned earlier, is significantly
+reduced network traffic and access to storage. Network traffic and access, often
+in the form of API access, are often a considerable cost factor, specifically
+also when hosted in public cloud provider systems.
+
+## Configuration
+
+Use the properties from the following table in your catalog properties files to
+enable and configure caching for the specific catalogs.
+
+:::{list-table} Cache configuration properties
+:widths: 25, 75
+:header-rows: 1
+
+* - Property
+  - Description
+* - `fs.cache`
+  - Configure the caching type. Defaults to no caching with the value `none`.
+    Set to `alluxio` to enable object storage caching powered by Alluxio.
+* - `fs.cache.directories`
+  - Required, comma-separated list of absolute paths to directories to use for
+    caching. All directories must exist on the coordinator and all workers.
+    Trino must have read and write permissions for files and nested directories.
+    A valid example with only one directory is `/tmp/trino-cache`.
+* - `fs.cache.max-sizes`
+  - Optional, comma-separated list of maximum [data sizes](prop-type-data-size)
+    for each caching directory. Order of values must be identical to the
+    directories list. Can not be used together with
+    `fs.cache.max-disk-usage-percentages`.
+* - `fs.cache.max-disk-usage-percentages`
+  - Optional, comma-separated list of maximum percentage values of the used disk
+    for each directory. Each value is an integer between 1 and 100. Order of
+    values must be identical to the directories list. If multiple directories
+    use the same disk, ensure that total percentages per drive remains below 100
+    percent. Can not be used together with `fs.cache.max-sizes`.
+* - `fs.cache.ttl`
+  -  The maximum [duration](prop-type-duration) for objects to remain in the cache
+     before eviction. Defaults to `7d`. The minimum value of `0s` means that caching
+     is effectively turned off.
+* - `fs.cache.preferred-hosts-count`
+  - The number of preferred nodes for caching files. Defaults to 2. Processing
+    identifies and subsequently prefers using specific nodes. If a query
+    requires more nodes, additional nodes are used after the preferred nodes.
```

Comment:
This description is a little confusing imho, it reads a little like "files will be cached on two nodes, and all other nodes in my cluster won't be caching files".

Unless I'm mistaken, this setting controls how much we distribute caching, between 1 (only 1 copy of the file is cached, on a single node) and $NUM_WORKERS (where all files are cached on all workers = cache is not distributed)?

Maybe we could add a documentation section about how we do distributed caching, which allows an effective cache size of (number of nodes) * (cache size) / (preferred-hosts-count), and it'll become clearer that this config option will allow more duplication to help with node failures / spreading workload, at the expense of effective cache size 👍

Reply:
This makes sense as an explanation and I have updated the wording some more. However I kinda questions the choice of having this configuration at all. It seems to artificially skew caching to a subset of the workers. They will receive more workload and more cache churn. Why would we not recommend setting this value to the number of workers in the cluster? Or even get rid of it and automatically cache equally on all nodes? Thoughts @electrum @raunaqmorarka @Pluies @jkylling ?